vcf/compare.go

```diff
@@ -118,39 +67,76 @@ func CompareAlt(alpha []string, beta []string) int {
 
 //Sort sorts a slice of Vcf structs in place.
 func Sort(vcfFile []Vcf) {
-	sort.Slice(vcfFile, func(i, j int) bool { return CompareVcf(vcfFile[i], vcfFile[j]) == -1 })
+	sort.Slice(vcfFile, func(i, j int) bool { return compareVcf(vcfFile[i], vcfFile[j]) == -1 })
 }
 
 //isEqual returns true if two input Vcf structs contain identical information, false otherwise.
 func isEqual(alpha Vcf, beta Vcf) bool {
-	if strings.Compare(alpha.Chr, beta.Chr) != 0 {
+	if alpha.Chr != beta.Chr {
 		return false
 	}
 	if alpha.Pos != beta.Pos {
 		return false
 	}
-	if strings.Compare(alpha.Id, beta.Id) != 0 {
+	if alpha.Id != beta.Id {
 		return false
 	}
-	if strings.Compare(alpha.Ref, beta.Ref) != 0 {
+	if alpha.Ref != beta.Ref {
 		return false
 	}
 	if CompareAlt(alpha.Alt, beta.Alt) != 0 {
 		return false
 	}
-	if strings.Compare(alpha.Filter, beta.Filter) != 0 {
+	if alpha.Filter != beta.Filter {
 		return false
 	}
-	if strings.Compare(alpha.Info, beta.Info) != 0 {
+	if alpha.Info != beta.Info {
 		return false
 	}
-	if CompareGenomeSamples(alpha.Samples, beta.Samples) != 0 {
+	if !equalSamples(alpha.Samples, beta.Samples) {
 		return false
 	}
 	return true
 
 }
 
+// equalSamples returns true if input samples are equivalent
+func equalSamples(alpha, beta []Sample) bool {
+	if len(alpha) != len(beta) {
+		return false
+	}
+	for i := range alpha {
+		if len(alpha[i].Alleles) != len(beta[i].Alleles) {
+			return false
+		}
+		if len(alpha[i].Phase) != len(beta[i].Phase) {
+			return false
+		}
+		if len(alpha[i].FormatData) != len(beta[i].FormatData) {
+			return false
+		}
+
+		for j := range alpha[i].Alleles {
+			if alpha[i].Alleles[j] != beta[i].Alleles[j] {
+				return false
+			}
+		}
+
+		for j := range alpha[i].Phase {
+			if alpha[i].Phase[j] != beta[i].Phase[j] {
+				return false
+			}
+		}
+
+		for j := range alpha[i].FormatData {
+			if alpha[i].FormatData[j] != beta[i].FormatData[j] {
+				return false
+			}
+		}
+	}
+	return true
+}
+
 //AllEqual returns true if each Vcf in a slice of Vcf structs contain identical information.
 func AllEqual(alpha []Vcf, beta []Vcf) bool {
```

Comment:
When comparing 2 unphased vcfs, I imagine it might be possible for vcf1 and vcf2 to share allele[0] and allele[1], but they are just swapped in position so allele[0] in vcf1 is allele[[1] in vcf2 and vice vera. Does this happen in vcfs? If so, then does the current gonomics code consider this?

Reply:
True. I think this is mostly (entirely?) used for testing at this point so I can inclined to leave it for now.